common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -124,7 +125,15 @@ public ShuffleIndexInformation load(String filePath) throws IOException {
       .weigher((Weigher<String, ShuffleIndexInformation>)
         (filePath, indexInfo) -> indexInfo.getRetainedMemorySize())
       .build(indexCacheLoader);
-    db = LevelDBProvider.initLevelDB(this.registeredExecutorFile, CURRENT_VERSION, mapper);
+    DBBackend dbBackend = null;
+    if (registeredExecutorFile != null) {
+      String dbBackendName =
+        conf.get(Constants.SHUFFLE_SERVICE_DB_BACKEND, DBBackend.LEVELDB.name());
+      dbBackend = DBBackend.byName(dbBackendName);
+      logger.info("Configured {} as {} and actually used value {}",
+        Constants.SHUFFLE_SERVICE_DB_BACKEND, dbBackendName, dbBackend);
+    }
+    db = DBProvider.initDB(dbBackend, this.registeredExecutorFile, CURRENT_VERSION, mapper);
```

Comment:
Ideally, this should be put inside the if block?

Reply:
Yes, you're right. I'll give a follow-up to fix this

Reply:
Need change to 

```java
if (registeredExecutorFile != null) {
      ....
      db = DBProvider.initDB(dbBackend, this.registeredExecutorFile, CURRENT_VERSION, mapper);
    } else {
      db = null;
    }
```
due to `db` is defined as final, and when 'registeredExecutorFile' is null, 'DBProvider.initDB' will return null directly . So  does `DBProvider.initDB` still need to be put inside the if block ?

Reply:
If so, I think `if (registeredExecutorFile != null)` is actually useless here..

Reply:
fixed

Reply:
`DBProvider.initDB` handles `null` `registeredExecutorFile` as input.

Reply:
I'm trying to delete `if (registeredExecutorFile != null)` condition

Reply:
The formulation in #37648 looks ok to me.